src/index.ts

```diff
@@ -20,12 +34,53 @@ export const rules = {
 	"valid-version": validVersion,
 };
 
+const recommendedRuleSettings = Object.fromEntries(
+	Object.entries(rules)
+		.filter(([, rule]) => rule.meta.docs?.recommended)
+		.map(([name]) => ["package-json/" + name, "error" as const]),
+);
+
+interface Configs {
+	recommended: {
+		rules: typeof recommendedRuleSettings;
+	};
+	"recommended-flat": {
+		files: string[];
+		languageOptions: {
+			parser: typeof parserJsonc;
+		};
+		plugins: {
+			"package-json": typeof plugin;
+		};
+		rules: typeof recommendedRuleSettings;
+	};
+}
+
+const plugin = {
+	configs: {} as Configs,
+	meta: {
+		name,
+		version,
+	},
+	rules,
+};
+
 export const configs = {
 	recommended: {
-		rules: Object.fromEntries(
-			Object.entries(rules)
-				.filter(([, rule]) => rule.meta.docs?.recommended)
-				.map(([name]) => ["package-json/" + name, "error" as const]),
-		),
+		rules: recommendedRuleSettings,
+	},
+	["recommended-flat"]: {
+		files: ["**/package.json"],
+		languageOptions: {
+			parser: parserJsonc,
+		},
+		plugins: {
+			"package-json": plugin,
```

Comment:
[Refactor] Heh, I think I see why you had to alternate between declaring `configs` and `plugins`. It's that there's a circular reference between `configs.plugins["recommended-flat"]["package-json"]` and `plugins.configs` right?

We can get around that with a getter:

 ```suggestion
				get "package-json"() {
				  return plugin;
				}
```

No more need for an `interface`, `as` assertion, or `Object.assign`. 😎